packages/client/hmi-client/src/workflow/ops/model-config/tera-parameter-table.vue

```diff
@@ -245,39 +284,153 @@ const typeOptions = [
 ];
 const props = defineProps<{
 	modelConfiguration: ModelConfiguration;
-	data: ModelConfigTableData[];
+	data?: ModelConfigTableData[]; // we can use our own passed in data or the computed one.  this is for the embedded matrix table
 	hideHeader?: boolean;
 }>();
 
 const emit = defineEmits(['update-value', 'update-configuration']);
 
 const matrixModalContext = ref({
 	isOpen: false,
-	stratifiedMatrixType: StratifiedMatrix.Initials,
 	matrixId: ''
 });
 
+const parameters = computed<Map<string, string[]>>(() => {
+	const model = props.modelConfiguration.configuration;
+	if (stratifiedModelType.value) {
+		return getUnstratifiedParameters(model);
+	}
+	const result = new Map<string, string[]>();
+	if (modelType.value === AMRSchemaNames.PETRINET || modelType.value === AMRSchemaNames.STOCKFLOW) {
+		model.semantics?.ode.parameters?.forEach((p) => {
+			result.set(p.id, [p.id]);
+		});
+	} else if (modelType.value === AMRSchemaNames.REGNET) {
+		model.model.parameters?.forEach((p) => {
+			result.set(p.id, [p.id]);
+		});
+	}
+	return result;
+});
+
+const getParamType = (param: ModelParameter | undefined) => {
+	let type = ParamType.CONSTANT;
+	if (!param) return type;
+	if (
+		props.modelConfiguration.configuration.metadata?.timeseries?.[param.id] ||
+		props.modelConfiguration.configuration.metadata.timeseries?.[param.id] === ''
+	) {
+		type = ParamType.TIME_SERIES;
+	} else if (param?.distribution) {
+		type = ParamType.DISTRIBUTION;
+	}
+	return type;
+};
+
+const tableFormattedParams = computed<ModelConfigTableData[]>(() => {
+	const formattedParams: ModelConfigTableData[] = [];
+
+	if (stratifiedModelType.value) {
+		parameters.value.forEach((vals, init) => {
+			const tableFormattedMatrix: ModelConfigTableData[] = vals.map((v) => {
+				let param;
+				if (modelType.value === AMRSchemaNames.REGNET) {
+					param = props.modelConfiguration.configuration.model.parameters.find((i) => i.id === v);
+				} else {
+					param = props.modelConfiguration.configuration.semantics.ode.parameters.find(
+						(i) => i.id === v
+					);
+				}
+				const paramType = getParamType(param);
+				const timeseriesValue =
+					props.modelConfiguration.configuration.metadata?.timeseries[param!.id];
+				const parametersMetadata =
+					props.modelConfiguration.configuration.metadata?.parameters?.[param!.id];
```

Comment:
It is easier to read if you don't repeat `props.modelConfiguration.configuration` so much, it is repeated 8 or 9 times here, can we just put that into a variable at the top of the function?

Reply:
done 😄